src/main/java/io/lakefs/routerfs/PathMapper.java

```diff
@@ -36,20 +41,34 @@ enum MappingConfigType {
     }
 
     private List<PathMapping> pathMappings;
-    private final PathMapping defaultMapping;
+    private final List<PathMapping> defaultMappings;
 
-    public PathMapper(Configuration conf, String defaultFromScheme, String defaultToScheme) throws IOException {
-        Objects.requireNonNull(defaultFromScheme);
-        Objects.requireNonNull(defaultToScheme);
+    public PathMapper(Configuration conf, @NonNull List<DefaultSchemeTranslation> defaultSchemeTranslations) throws IOException {
         this.pathMappings = new ArrayList<>();
-        this.defaultMapping = createDefaultMapping(defaultFromScheme, defaultToScheme);
+        if(defaultSchemeTranslations.isEmpty()) {
```

Comment:
```

```suggestion
        if (defaultSchemeTranslations.isEmpty()) {
```

```